packages/devextreme/js/__internal/grids/grid_core/header_filter/m_header_filter.ts

```diff
@@ -418,15 +418,18 @@ const ColumnHeadersViewHeaderFilterExtender = extend({}, headerFilterMixin, {
 
   _columnOptionChanged(e) {
     const { optionNames } = e;
+    const isFilterRowAndHeaderFilterValuesChanged = gridCoreUtils.checkChanges(optionNames, ['filterValues', 'filterValue']);
+    const isHeaderFilterValuesAndTypeChanged = gridCoreUtils.checkChanges(optionNames, ['filterValues', 'filterType']);
 
-    if (gridCoreUtils.checkChanges(optionNames, ['filterValues', 'filterType'])) {
+    if (isFilterRowAndHeaderFilterValuesChanged || isHeaderFilterValuesAndTypeChanged) {
       if (this._needUpdateFilterIndicators()) {
         this._updateHeaderFilterIndicators();
       }
```

Comment:
Minor comment.

Let's get rid of inner ```if``` too.
Something like this:
```suggestion
    const shouldUpdateFilterIndicators = (
      isFilterRowAndHeaderFilterValuesChanged || isHeaderFilterValuesAndTypeChanged
    ) && this._needUpdateFilterIndicators();
    
    if (shouldUpdateFilterIndicators) {
      this._updateHeaderFilterIndicators();
   }
```